### DISCUSSION TITLE
Path parameters vs Query arguments.

### DISCUSSION THREAD

**Discussion by bobmcwhirter** [guacsec/trustify#122] [Tenets]
Prefer query args? 

@jcrossley3

**Comment by ctron** [comment]
I noticed the upload function. If feels weird having to provide the "format" as part of the query during the upload. I think in this case I would prefer a path argument.

If think query arguments should be for … queries. Or for options/flags which are optional.

**Reply by bobmcwhirter** [reply]
so, POST, avoid query args unless they're optional.
for GET, where args might filter/flag things, they make sense.
